mm/mm_heap/mm_initialize.c

```diff
@@ -93,12 +93,17 @@ void mm_addregion(FAR struct mm_heap_s *heap, FAR void *heapstart,
 
   DEBUGVERIFY(mm_takesemaphore(heap));
 
-  /* Adjust the provided heap start and size so that they are both aligned
-   * with the MM_MIN_CHUNK size.
+  /* Adjust the provided heap start and size.
+   *
+   * Note: (uintptr_t)node + SIZEOF_MM_ALLOCNODE is what's actually
+   * returned to the malloc user, which should have natural alignment.
+   * (that is, in this implementation, MM_MIN_CHUNK-alignment.)
    */
 
-  heapbase = MM_ALIGN_UP((uintptr_t)heapstart);
-  heapend  = MM_ALIGN_DOWN((uintptr_t)heapstart + (uintptr_t)heapsize);
+  heapbase = MM_ALIGN_UP((uintptr_t)heapstart + SIZEOF_MM_ALLOCNODE)
+             - SIZEOF_MM_ALLOCNODE;
+  heapend  = MM_ALIGN_DOWN((uintptr_t)heapstart + (uintptr_t)heapsize)
```

Comment:
Do we still need this patch? https://github.com/apache/incubator-nuttx/pull/5906 should already fix this issue.

Reply:
yes.
i'm sure i tested recent enough version containing https://github.com/apache/incubator-nuttx/pull/5906

Reply:
It's strange that the original code:

1. Align heapbase and heapend to MM_MIN_CHUNK
2. #5906 ensure two preallocate node with MM_MIN_CHUNK

So, the rest of free slot should multiple of MM_MIN_CHUNK.

Reply:
@CV-Bowen
even after this change, (heapend - heapbase) % MM_MIN_CHUNK == 0.
there is no problem wrt the goal of #5906 as far as i understand.

Reply:
@xiaoxiang781216 @yamt we can satisfy the goal of https://github.com/apache/incubator-nuttx/pull/5991 and https://github.com/apache/incubator-nuttx/pull/5906 at the same time. Yesterday i missed some code and came to a wrong conclusion.

Reply:
But, look like #5906 isn't need anymore?